src/seedu/addressbook/AddressBook.java

```diff
@@ -923,12 +925,35 @@ private static String encodePersonToString(String[] person) {
         final String[] decodedPerson = makePersonFromData(
                 extractNameFromPersonString(encoded),
                 extractPhoneFromPersonString(encoded),
-                extractEmailFromPersonString(encoded)
+                extractEmailFromPersonString(encoded),
+                extractDobFromPersonString(encoded)
         );
         // check that the constructed person is valid
         return isPersonDataValid(decodedPerson) ? Optional.of(decodedPerson) : Optional.empty();
     }
 
+    private static String extractDobFromPersonString(String encoded) {
+        final int indexOfPhonePrefix = encoded.indexOf(PERSON_DATA_PREFIX_PHONE);
+        final int indexOfEmailPrefix = encoded.indexOf(PERSON_DATA_PREFIX_EMAIL);
+        final int indexOfDobPrefix = encoded.indexOf("d/");
+
+        // phone is last arg, target is from prefix to end of string
+        if ((indexOfDobPrefix > indexOfEmailPrefix) && (indexOfDobPrefix > indexOfPhonePrefix)) {
+            return removePrefix(encoded.substring(indexOfPhonePrefix, encoded.length()).trim(),
+                    PERSON_DATA_PREFIX_PHONE);
+
+        // phone is middle arg, target is from own prefix to next prefix
+        } else if ((indexOfDobPrefix >indexOfEmailPrefix) && (indexOfDobPrefix < indexOfPhonePrefix)) {
+            return removePrefix(
+                    encoded.substring(indexOfDobPrefix, indexOfPhonePrefix).trim(),
+                    PERSON_DATA_PREFIX_PHONE);            
+        } else {
+            return removePrefix(
+                    encoded.substring(indexOfDobPrefix, indexOfEmailPrefix).trim(),
```

Comment:
Indentation problem here?. Our coding standard requires you to use 4 spaces instead of tabs. You can configure Eclipse to convert tabs to spaces automatically.